sklearn/cluster/_dbscan_inner.pyx

```diff
@@ -5,14 +5,12 @@
 from libcpp.vector cimport vector
 cimport numpy as cnp
 
-cnp.import_array()
 
-
-def dbscan_inner(cnp.ndarray[cnp.uint8_t, ndim=1, mode='c'] is_core,
-                 cnp.ndarray[object, ndim=1] neighborhoods,
-                 cnp.ndarray[cnp.npy_intp, ndim=1, mode='c'] labels):
+def dbscan_inner(cnp.uint8_t[::1] is_core,
+                 object[:] neighborhoods,
+                 cnp.npy_intp[::1] labels):
     cdef cnp.npy_intp i, label_num = 0, v
-    cdef cnp.ndarray[cnp.npy_intp, ndim=1] neighb
+    cdef cnp.npy_intp[:] neighb
```

Comment:
this doesn't have negative performance implications, does it?

Reply:
Do I need to do provide some benchmark?

Reply:
@jjerphan seems to be confident these are actually faster. In which case I don't think we need a benchmark :)

Reply:
I did a quick profiling and I could not spot any difference.

Reply:
For reference, more information about the usage of NumPy arrays and of memoryviews in Cython are given in [this section of the documentation](https://cython.readthedocs.io/en/latest/src/tutorial/numpy.html?highlight=numpy#working-with-numpy).